docs/guides/devops-lifecycle/integrate-web-modeler-in-ci-cd.md

```diff
@@ -0,0 +1,332 @@
+
+Progress is underway to introduce webhook registration or event subscription for milestone creation monitoring.
+
+Combine these two approaches and listen to milestones to sync files to your version control, create a pull/merge request, and trigger pipelines.
+
+## Pipeline stages
+
+The following examples illustrate setting up **build**, **test**, **review**, and **publish** stages within a pipeline.
+
+### Build stage
+
+While there is no distinct concept for a build package in Camunda Platform 8, artifact structuring depends on your overall software architecture. The build stage should primarily focus on acquiring dependencies and deploying them to a preview environment.
+
+#### Set up preview environments
+
+Offering an automatically testable and review-ready process preview mandates a dedicated preview cluster. Numerous options exist, varying with software development lifecycle design, preferences, and Camunda Platform 8 deployment type (SaaS, self-managed, or hybrid). This guide proposes a setup with lightweight local self-managed preview clusters (or embedded engines) and full-fledged staging and production clusters (Self-Managed or SaaS).
+
+##### Using fully-featured clusters
+
+For local preview environments, you can deploy a comprehensive [Zeebe](https://github.com/camunda/zeebe) cluster including Operate and Tasklist. Options include using docker-compose or Kubernetes via Helm. All necessary endpoints and UIs are available for thorough process/application testing. Opt for a cluster version aligned with your production cluster to ensure process compatibility.
+
+##### Using embedded Zeebe engines
+
+If you don't need to spawn all apps such as Operate or Tasklist, you can use the lightweight [embedded Zeebe engine](https://github.com/camunda-community-hub/eze), which is a community-maintained project, to set up a cost-effective solution with an in-memory database. Together with the [Zeebe Hazelcast exporter](https://github.com/camunda-community-hub/zeebe-hazelcast-exporter) (community-maintained as well), you can consume data generated from your process for reporting or testing.
+
+In the build stage, deploy your process or project to a cluster or embedded engine. Post-pipeline completion, such as deployment to staging or production, preview environments can be discarded.
+
+:::tip
+For GitLab users, consider using [GitLab Review Apps](https://docs.gitlab.com/ee/ci/review_apps/) to provide preview environments.
+:::
+
+Deploy resources using the [`zbctl` CLI](/apis-tools/cli-client) in this pipeline step, compatible with both SaaS and Self-Managed clusters. Alternately, utilize the Java or Go client library or any community-built alternatives.
```

Comment:
Do we want to link to Java/Go client libraries here, and community clients as well?